app/models/ip_based_group.rb

```diff
@@ -0,0 +1,41 @@
+class IpBasedGroup < ActiveRecord::Base
+  belongs_to :role
+  validate :has_valid_ip_addresses
+
+  def self.for_ip(ip)
+    ip_int = IpConverter.ip_to_i(ip)
+    if ip_int
```

Comment:
Yeah, I would like to stay away from multiple returns. I'd much rather stick to rails' implicit return.